chapters/annotations.tex

```diff
@@ -2052,24 +2055,24 @@ \subsection{Protection of Classes}\label{protection-of-classes}
   Same as \lstinline!Access.packageText! and additionally the class, or part of the class, can be copied.
 \end{enumerate}
 
-The \lstinline!access! annotation holds for the respective class and all classes
-that are hierarchically on a lower level, unless overridden by a
-\lstinline!Protection! annotation with \lstinline!access!.
-Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate!
-has no effect.
+The \lstinline!access! annotation holds for the respective class and all classes that are hierarchically on a lower level, unless overridden by a \lstinline!Protection! annotation with \lstinline!access!.
+Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate! has no effect.
+
+Classes should not use other classes in ways that contradict this protection.
+For instance a class with \lstinline!Access.hide! should not be used in the diagram layer of a class with \lstinline!Access.diagram!, and there should not be hyperlinks to classes with \lstinline!Access.icon! (from classes with visible documentation).
```

Comment:
There should be no room for interpretation:
```suggestion
Classes must not use other classes in ways that contradict this protection.
For instance, a class with \lstinline!Access.hide! must not be used in the diagram layer of a class with \lstinline!Access.diagram!, and there must not be hyperlinks to classes with \lstinline!Access.icon! from classes with visible documentation.
```

Reply:
I wonder if we could go even further, and give the explicit rules that the generic formulation boils down to?
```suggestion
Classes must not use other classes in ways that contradict this protection:
\begin{itemize}
\item A class with \lstinline!Access.hide! or higher protection must not be used in the diagram layer of a class with \lstinline!Access.diagram!.
\item There must not be hyperlinks to classes with \lstinline!Access.icon! or higher protection from classes with visible documentation.
…
\end{itemize}
```

Reply:
I feared that the full list would be rather long, which would risk that something was missed and thus it was clearer to just give the goal.

The use of "should" instead of "must" was deliberate. If a package makes an error here that should not prevent us from simulating its models, so a strict requirement could be a problem.

Reply:
As a library author defending for my intellectual property, I'd be disappointed if the specification didn't strictly require tools to obey access restrictions.  If they aren't set up correctly, the library should be fixed so that there is no ambiguity about how a tool should handle the contents.

Reply:
I agree that tools should obey the access restrictions, but the above is stated as restrictions on the classes - not on the tools.

So, to clarify add 
"Tools must ensure that protected contents is not shown, even if classes do not meet this requirement."

To me this "must" only makes sense if the restriction for the classes use "should", since "must" is an absolute requirement and here we are specifying what happens if the requirement is broken.

Note that this doesn't describe how the situation is handled by tools as long as the contents isn't shown - for diagrams Dymola have depending on circumstances used either Crossed-out-icons and just no indication; but the classes with those diagrams could still be used. (Obviously the specific libraries were corrected later.)

Reply:
Sounds like a good discussion point for a web meeting.

Reply:
This was now added after the web-meeting.